helix-term/src/ui/document.rs

```diff
@@ -428,8 +324,46 @@ impl<'a> TextRenderer<'a> {
             text_style,
             draw_indent_guides: editor_config.indent_guides.render,
             viewport,
-            col_offset,
+            offset,
+        }
+    }
+    /// Draws a single `grapheme` at the current render position with a specified `style`.
+    pub fn draw_decoration_grapheme(
+        &mut self,
+        grapheme: Grapheme,
+        mut style: Style,
+        mut row: u16,
+        col: u16,
+    ) -> bool {
+        if (row as usize) < self.offset.row
+            || row >= self.viewport.height
+            || col >= self.viewport.width
+        {
+            return false;
+        }
+        row -= self.offset.row as u16;
+        // TODO is it correct to apply the whitspace style to all unicode white spaces?
+        if grapheme.is_whitespace() {
+            style = style.patch(self.whitespace_style);
         }
+
+        let grapheme = match grapheme {
+            Grapheme::Tab { width } => {
+                let grapheme_tab_width = char_to_byte_idx(&self.virtual_tab, width);
+                &self.virtual_tab[..grapheme_tab_width]
+            }
+            Grapheme::Other { ref g } if g == "\u{00A0}" => " ",
```

Comment:
```suggestion
            Grapheme::Other { ref g } if g == "\u{00A0}" | g == "\u{202F}" => " ",
```

for https://github.com/helix-editor/helix/pull/9604

or I wonder if `g.is_whitespace()` is more correct here?

Reply:
I am honestly not sure about this one. Really this function is not used for normal graphemes just decrations like inline diagnostics. Since we don't apply the configurable whitespace rendering to those I think it makes more sense to actually not do this replacement at all (just remove this line) so the text looks as close as possible to the original (and that this code stays simple).